sound/soc/codecs/pcm512x.c

```diff
@@ -1701,17 +1704,27 @@ int pcm512x_probe(struct device *dev, struct regmap *regmap)
 
 	gpiod_add_lookup_table(&pcm512x_gpios_table);
 
-	pcm512x->sclk = devm_clk_get(dev, NULL);
+retry_sclk_name:
+	pcm512x->sclk = devm_clk_get(dev, clk_name);
 	if (PTR_ERR(pcm512x->sclk) == -EPROBE_DEFER) {
 		ret = -EPROBE_DEFER;
 		goto err;
 	}
 	if (!IS_ERR(pcm512x->sclk)) {
+		dev_dbg(dev, "SCLK detected by devm_clk_get\n");
 		ret = clk_prepare_enable(pcm512x->sclk);
 		if (ret != 0) {
 			dev_err(dev, "Failed to enable SCLK: %d\n", ret);
 			goto err;
 		}
+	} else {
+		if (!clk_name) {
+			dev_dbg(dev, "no SCLK detected, retrying with fixed sclk ID\n");
+			clk_name = (char *)sclk_name;
+			goto retry_sclk_name;
```

Comment:
I would just do `const char *clk_name[] = {NULL, "sclk"};` and then use a loop instead of a `goto`

Reply:
ok, it's only marginally better but fine